master branch does not build with idle rate counters enabled

Use
```
HPX_WITH_THREAD_IDLE_RATES=ON
```
Error
```
[  0%] Built target hpx_wrap
[  0%] Built target hpx_preprocessor
[  0%] Built target hpx_init
[ 10%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/detail/scheduled_thread_pool.cpp.o
[ 10%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/thread.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/thread_data.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/thread_helpers.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/thread_executor.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/thread_pool_base.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/threadmanager.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/runtime/threads/topology.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/runtime/trigger_lco.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/performance_counters/counter_creators.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/performance_counters/counter_parser.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/performance_counters/counters.cpp.o
[ 20%] Building CXX object src/CMakeFiles/hpx.dir/performance_counters/manage_counter.cpp.o
In file included from /scratch/snx3000/biddisco/src/hpx/src/runtime/threads/detail/scheduled_thread_pool.cpp:7:0:
/scratch/snx3000/biddisco/src/hpx/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp:862:36: error: no 'int64_t hpx::threads::detail::scheduled_thread_pool<Scheduler>::get_thread_phase_duration(std::size_t, bool)' member function declared in class 'hpx::threads::detail::scheduled_thread_pool<Scheduler>'
         std::size_t num, bool reset)
                                    ^
/scratch/snx3000/biddisco/src/hpx/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp:927:36: error: no 'int64_t hpx::threads::detail::scheduled_thread_pool<Scheduler>::get_thread_duration(std::size_t, bool)' member function declared in class 'hpx::threads::detail::scheduled_thread_pool<Scheduler>'
         std::size_t num, bool reset)
                                    ^
/scratch/snx3000/biddisco/src/hpx/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp:989:36: error: no 'int64_t hpx::threads::detail::scheduled_thread_pool<Scheduler>::get_thread_phase_overhead(std::size_t, bool)' member function declared in class 'hpx::threads::detail::scheduled_thread_pool<Scheduler>'
         std::size_t num, bool reset)
                                    ^
/scratch/snx3000/biddisco/src/hpx/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp:1082:36: error: no 'int64_t hpx::threads::detail::scheduled_thread_pool<Scheduler>::get_thread_overhead(std::size_t, bool)' member function declared in class 'hpx::threads::detail::scheduled_thread_pool<Scheduler>'
         std::size_t num, bool reset)
                                    ^
/scratch/snx3000/biddisco/src/hpx/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp:1172:36: error: no 'int64_t hpx::threads::detail::scheduled_thread_pool<Scheduler>::get_cumulative_thread_duration(std::size_t, bool)' member function declared in class 'hpx::threads::detail::scheduled_thread_pool<Scheduler>'
         std::size_t num, bool reset)
                                    ^
/scratch/snx3000/biddisco/src/hpx/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp:1217:36: error: no 'int64_t hpx::threads::detail::scheduled_thread_pool<Scheduler>::get_cumulative_thread_overhead(std::size_t, bool)' member function declared in class 'hpx::threads::detail::scheduled_thread_pool<Scheduler>'
         std::size_t num, bool reset)
```                                    ^

--- a/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp
+++ b/hpx/runtime/threads/detail/scheduled_thread_pool_impl.hpp
@@ -854,7 +854,6 @@ namespace hpx { namespace threads { namespace detail
 
         return executed_phases - reset_executed_phases;
     }
-#endif
 
 #if defined(HPX_HAVE_THREAD_IDLE_RATES)
     template <typename Scheduler>
@@ -1279,8 +1278,8 @@ namespace hpx { namespace threads { namespace detail
         return std::int64_t(
             (double(tfunc_total) - double(exec_total)) * timestamp_scale_);
     }
-#endif
-#endif
+#endif // HPX_HAVE_THREAD_IDLE_RATES
+#endif // HPX_HAVE_THREAD_CUMULATIVE_COUNTS
 
 #if defined(HPX_HAVE_BACKGROUND_THREAD_COUNTERS) && defined(HPX_HAVE_THREAD_IDLE_RATES)
     ////////////////////////////////////////////////////////////
@@ -1530,7 +1529,7 @@ namespace hpx { namespace threads { namespace detail
             (cleanup_total / double(tfunc_total - exec_total));
         return std::int64_t(10000. * percent);    // 0.01 percent
     }
-#endif
+#endif // HPX_HAVE_THREAD_CREATION_AND_CLEANUP_RATES
 
     template <typename Scheduler>
     std::int64_t scheduled_thread_pool<Scheduler>::avg_idle_rate_all(bool reset)
@@ -1612,7 +1611,7 @@ namespace hpx { namespace threads { namespace detail
             1. - (double(exec_time) / double(tfunc_time));
         return std::int64_t(10000. * percent);    // 0.01 percent
     }
-#endif
+#endif // HPX_HAVE_THREAD_IDLE_RATES
 
     template <typename Scheduler>
     std::int64_t scheduled_thread_pool<Scheduler>::get_idle_loop_count(
@@ -2038,3 +2037,5 @@ namespace hpx { namespace threads { namespace detail
         }
     }
 }}}
+
+#endif

